smart_home/include/devices/device.h

```diff
@@ -0,0 +1,24 @@
+#ifndef SMART_HOME_DEVICES_DEVICE_H_
+#define SMART_HOME_DEVICES_DEVICE_H_
+
+#include "../rooms/room.h"
+
+namespace devices {
+
+// Rule of Five
```

Comment:
rule of 5 is not implemented in this class.
For a proper rule of 5 you need to define all 5 functions (Copy,Move Constructors, Copy,Move assignment operators and Destructor)